Safety-Profile: Add FunctionalSafety profile with Verification and Evidence classes

nicpappler opened the pull request:
- Created new folder  with Classes, Properties, and Vocabularies
- added classes
- Defined supporting properties (evaluation, rationale, UUIDs, categories, pre/postconditions, methods)
- Updated Core vocabularies 

Content already discussed on the safety-profile fork

nicpappler committed:
Add FunctionalSafety profile with Verification and Evidence classes

- Created new folder  with Classes, Properties, and Vocabularies
- Added , , and  classes
- Defined supporting properties (evaluation, rationale, UUIDs, categories, pre/postconditions, methods)
- Introduced vocabularies for , , and
- Updated Core vocabularies (, ) to support safety extensions

bact commented:
spec-parser failed:

> Missing top-level namespace file FunctionalSafety.md

Need the FunctionalSafety.md in FunctionalSafety/ directory.

bact commented on model/Core/Vocabularies/RelationshipType.md:
```diff
@@ -25,35 +25,29 @@ name completes the sentence:
 - copiedTo: The `from` Element has been copied to each `to` Element.
-- createdBy: The `from` Element's Action or DefinedProcess is createdBy `to` Agent(s).
-- delegatedTo: The `from` Agent is delegating an action to the Agent of the `to` Relationship (which shall be of type invokedBy), during a LifecycleScopeType (e.g. the `to` invokedBy Relationship is being done on behalf of `from`).
+- delegatedTo: The `from` Agent is delegating an action to the Agent of the `to` Relationship (which must be of type invokedBy), during a LifecycleScopeType (e.g. the `to` invokedBy Relationship is being done on behalf of `from`).
```
As this will be part of the spec, please use ISO verbal forms (e.g. "shall" for requirement, "should" for recommendation). Avoid "must".

See: https://github.com/spdx/spdx-3-model/blob/develop/docs/format.md#verbal-forms-for-expressions-of-provisions

bact commented on model/SystemsEngineering/Properties/verificationPostcondition.md:
```diff
@@ -0,0 +1,17 @@
+
+## Summary
+
+Verification postcondition that must be true immediately after a verification method has been performed
```
As this will be part of the spec, please use ISO verbal forms (e.g. "shall" for requirement, "should" for recommendation). Avoid "must".

See: https://github.com/spdx/spdx-3-model/blob/develop/docs/format.md#verbal-forms-for-expressions-of-provisions

bact commented on model/SystemsEngineering/Properties/verificationPostcondition.md:
```diff
@@ -0,0 +1,17 @@
+
+## Description
+
+Verification postcondition are explicit criteria, states, or conditions that must be true immediately after a verification method (such as a test, inspection, or analysis) has been performed, confirming the successful execution and complete outcomes of the verification.
```
Please use ISO verbal forms. Avoid "must".

nicpappler replied:
"Shall" does not express what we need to say here -> changed the must to something less controversial.

bact commented on model/SystemsEngineering/Properties/verificationPrecondition.md:
```diff
@@ -0,0 +1,17 @@
+
+## Description
+
+Verification preconditions are specific conditions, requirements, or initial states that must be satisfied before a verification will be executed. 
```
Please use ISO verbal forms. Avoid "must".

nicpappler replied:
"Shall" does not express what we need to say here -> changed the must to something less controversial.

bact commented on model/Core/Vocabularies/RelationshipType.md:
```diff
@@ -25,35 +25,29 @@ name completes the sentence:
 - dependsOn: The `from` Element depends on each `to` Element, during a LifecycleScopeType period.
 - descendantOf: The `from` Element is a descendant of each `to` Element.
-- describes: The `from` Element describes each `to` Element. To denote the root(s) of a tree of elements in a collection, the rootElement property shall be used.
+- describes: The `from` Element describes each `to` Element. To denote the root(s) of a tree of elements in a collection, the rootElement property should be used.
```
Since we are using ISO verbal form,
are you suggesting to change from a requirement to a recommendation?

nicpappler replied:
Issue lays in outdated base, will fix this.

bact commented on model/Core/Vocabularies/RelationshipType.md:
```diff
@@ -25,35 +25,29 @@ name completes the sentence:
 - doesNotAffect: The `from` Vulnerability has no impact on each `to` Element. The use of the `doesNotAffect` is constrained to `VexNotAffectedVulnAssessmentRelationship` classed relationships.
-- evaluatedOn: The `from` Element has been evaluated on the `to` Element(s).
-- expandsTo: The `from` Element expands out as an artifact described by each `to` Element.
+- expandsTo: The `from` archive expands out as an artifact described by each `to` Element.
```
We just recently specified a type for `from` in the merged PR https://github.com/spdx/spdx-3-model/pull/1123.

Is it possible that this PR is using an outdated base?

nicpappler replied:
yes, right.

bact commented on model/Core/Vocabularies/ExternalIdentifierType.md:
```diff
@@ -16,22 +16,17 @@ ExternalIdentifierType specifies the type of an external identifier.
-- cpe22: [Common Platform Enumeration Specification 2.2](https://cpe.mitre.org/files/cpe-specification_2.2.pdf).
-- cpe23: [Common Platform Enumeration: Naming Specification Version 2.3](https://csrc.nist.gov/publications/detail/nistir/7695/final).
+- cpe22: [Common Platform Enumeration Specification 2.2](https://cpe.mitre.org/files/cpe-specification_2.2.pdf)
+- cpe23: [Common Platform Enumeration: Naming Specification Version 2.3](https://csrc.nist.gov/publications/detail/nistir/7695/final)
```
```suggestion
- cpe22: [Common Platform Enumeration Specification 2.2](https://cpe.mitre.org/files/cpe-specification_2.2.pdf).
- cpe23: [Common Platform Enumeration: Naming Specification Version 2.3](https://csrc.nist.gov/publications/detail/nistir/7695/final).
```

Other entries have the fullstop.

bact commented on model/FunctionalSafety/Vocabularies/EvidenceType.md:
```diff
@@ -0,0 +1,23 @@
+
+## Summary
+
+EvidenceType refers to categories of documented or observable proof used to verify compliance, qualification, or performance 
```
```suggestion
EvidenceType refers to categories of documented or observable proof used to verify compliance, qualification, or performance.
```

nicpappler committed:
Create FunctionalSafety.md

adding the md file for the profile

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Adding the md file for the safety

Adding the content of the md file describing the Safety Profile

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

stanislaw committed:
model: rename: FunctionalSafety -> SystemsEngineering

nicpappler committed:
Merge pull request #4 from stanislaw/add-functional-safety

 model: rename: FunctionalSafety -> SystemsEngineering

nicpappler committed:
Create SystemsEngineering.md

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Enhance SystemsEngineering.md with detailed sections

Added summary, description, metadata, and profile conformance sections to the Systems Engineering documentation.

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update Safety metadata ID to version 3.1

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

bact commented:
Please make sure that this PR will not revert changes/additions by the Hardware and SupplyChain profiles, also the edits for ISO submission (for example, "this specification" --> "this document").

Check if this PR is based on the latest develop branch.

nicpappler committed:
Fix wording in verification postcondition documentation

must be -> are

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update wording in verificationPrecondition.md

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

stanislaw committed:
Merge remote-tracking branch 'origin/develop' into add-functional-safety

nicpappler committed:
Merge pull request #5 from stanislaw/add-functional-safety

Synchronize with the latest develop branch

stanislaw committed:
Fix model/Core/Vocabularies/ExternalIdentifierType.md

nicpappler committed:
Merge pull request #6 from stanislaw/add-functional-safety

Fix model/Core/Vocabularies/ExternalIdentifierType.md

goneall commented on model/SystemsEngineering/Properties/evidenceUUID.md:
```diff
@@ -0,0 +1,17 @@
+## Metadata
+
+- name: evidenceUUID
+- Nature: DataProperty
```
Is there an inconsistency between the Range being an `ExternalIdentifier` and the Nature being a `DataProperty`?

@zvr - thoughts?

stevenc-stb replied:
```suggestion
- Nature: ObjectProperty
```
good catch!

goneall commented on model/SystemsEngineering/Properties/evidenceCategories.md:
```diff
@@ -0,0 +1,17 @@
+## Metadata
+
+- name: evidenceCategories
+- Nature: DataProperty  
```
Possible inconsistency between the Nature and the Range

stevenc-stb replied:
```suggestion
- Nature: ObjectProperty
```

goneall commented on model/SystemsEngineering/Properties/verificationUUID.md:
```diff
@@ -0,0 +1,17 @@
+## Metadata
+
+- name: verificationUUID
+- Nature: DataProperty
```
Possible inconsistency between the Nature and the Range

stevenc-stb replied:
```suggestion
- Nature: ObjectProperty
```

goneall commented on model/SystemsEngineering/Properties/verifiedUsing.md:
```diff
@@ -0,0 +1,22 @@
+SPDX-License-Identifier: Community-Spec-1.0
```
This is already in core - please update to use the Core definition.

stevenc-stb replied:
This looks like it can be removed; it looks like it was unintentionally copied from core. It is not a new property in EvaluationResult, EvidenceRelationship, or RequirementVerification.”

nicpappler replied:
Yes, moved over all that seemed to belong to the verification stuff from one of the outdated PRs, and this one slipped in.

goneall commented on model/SystemsEngineering/Safety.md:
```diff
@@ -0,0 +1,23 @@
+- name: Safety
+
+## Profile conformance
+TBD
```
Just noting this needs to be filled in

nicpappler replied:
Removed this section as the other profiles I checked don't have this section.

goneall commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,22 @@
+- name: SystemsEngineering
+
+## Profile conformance
+- for every `/SystemsEngineering/RequirementVerification` there shall exist at least one `/Core/Relationship` of type `verifiedBy` having that element as its to property and a `/Core/Requirement` as its from property.
```
```suggestion
- for every `/SystemsEngineering/RequirementVerification` there shall exist at least one `/Core/Relationship` of relationship type `verifiedBy` having that element as its to property and a `/Core/Requirement` as its from property.
```

goneall commented on model/SystemsEngineering/Properties/verifiedUsing.md:
```diff
@@ -0,0 +1,22 @@
+
+- name: verifiedUsing
+- Nature: ObjectProperty
+- Range: IntegrityMethod
```
```suggestion
- Range: Core/IntegrityMethod
```

goneall commented on model/SystemsEngineering/Classes/EvidenceRelationship.md:
```diff
@@ -0,0 +1,27 @@
+## Properties
+
+- evidenceUUID
+  - type: ExternalIdentifier
```
```suggestion
  - type: Core/ExternalIdentifier
```

goneall commented on model/SystemsEngineering/Classes/RequirementVerification.md:
```diff
@@ -0,0 +1,39 @@
+## Properties
+
+- verificationUUID
+  - type: ExternalIdentifier
```
```suggestion
  - type: Core/ExternalIdentifier
```

goneall reviewed (changes requested):
Found a few possible issues - comments in the files

nicpappler committed:
Update model/SystemsEngineering/Properties/evidenceUUID.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Properties/verificationUUID.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/SystemsEngineering.md

Co-authored-by: Gary O'Neall <gary@sourceauditor.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Properties/verifiedUsing.md

Co-authored-by: Gary O'Neall <gary@sourceauditor.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Classes/EvidenceRelationship.md

Co-authored-by: Gary O'Neall <gary@sourceauditor.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Classes/RequirementVerification.md

Co-authored-by: Gary O'Neall <gary@sourceauditor.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Remove 'Profile conformance' section from Safety.md

Removed the 'Profile conformance' section and its placeholder.

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Properties/evidenceCategories.md

Co-authored-by: stevenc-stb <steven@smarttalkbeacon.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

bact commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,22 @@
+
+## Summary
+
+The Systems Engineering Profile is designed to provide a standardized way of documenting and
```
```suggestion
The Systems Engineering profile is designed to provide a standardized way of documenting and
```

Use lowercase "profile". To be consistent with other profiles.

nicpappler committed:
Delete model/SystemsEngineering/Properties/verifiedUsing.md

copy/paste issue

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

bact commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,21 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# Safety
```
This has to be removed. The profile name is changed.
We use SystemsEngineering.md now.

nicpappler replied:
I'd rather stash the SystemsEngineering.md now as we decided to keep the Safety Profile for 3.1 and add SystemsEngineering as an overall concept in 3.2

@kestewart  - can I get your opinion on that?

bact replied:
Got it. In that case, the `Safety.md` file should be inside `model/Safety/` directory.

kestewart replied:
Yes, would like to keep Safety for 3.1 and then consider SystemsEngeineering.md for 3.2

stevenc-stb replied:
i thought that Both SystemsEngeineering Profile and Safety Profile in 3.1. with the SystemsEngeineering having the the current class.
It was my impression that System Engineering was the parent to safety

> I'd rather stash the SystemsEngineering.md now as we decided to keep the Safety Profile for 3.1 and add SystemsEngineering as an overall concept in 3.2
> 
> @kestewart - can I get your opinion on that?

nicpappler committed:
Update model/SystemsEngineering/SystemsEngineering.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

bact commented on model/SystemsEngineering/Properties/verificationUUID.md:
```diff
@@ -0,0 +1,17 @@
+
+- name: verificationUUID
+- Nature: ObjectProperty
+- Range: ExternalIdentifier
```
```suggestion
- Range: /Core/ExternalIdentifier
```

bact commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,22 @@
+## Summary
+
+The Systems Engineering profile is designed to provide a standardized way of documenting and
+sharing information about artefacts created, verified and maintained during a system's lifecycle.
```
```suggestion
sharing information about artifacts created, verified and maintained during a system's lifecycle.
```

We use American-spelling "artifact" in other places.

bact commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,22 @@
+
+## Description
+
+The Systems Engineering profile's namespace defines a set of concepts and data elements related to artefacts and their dependencies or links to each other that are part of engineering related documentation.
```
```suggestion
The Systems Engineering profile's namespace defines a set of concepts and data elements related to artifacts and their dependencies or links to each other that are part of engineering related documentation.
```

bact commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,22 @@
+The Systems Engineering profile's namespace defines a set of concepts and data elements related to artefacts and their dependencies or links to each other that are part of engineering related documentation.
+These artifacts include the inputs and outputs of the systems engineering phases regarding planning, requirements analysis, system/software architecture, implemenation, and the verification tasks. 
+
+
```
```suggestion

```

bact commented on model/SystemsEngineering/Safety.md:
```diff
@@ -0,0 +1,21 @@
+## Summary
+
+The Safety Profile is designed to provide a standardized way of documenting and
+sharing information about artefacts created, verified and maintained during the safety lifecycle of a system.
```
```suggestion
sharing information about artifacts created, verified and maintained during the safety lifecycle of a system.
```

We use American-spelling "artifact" elsewhere.

bact commented on model/SystemsEngineering/Safety.md:
```diff
@@ -0,0 +1,21 @@
+
+## Description
+
+The safety profile's namespace defines a set of concepts and data elements related to artefacts and their dependencies or links
```
```suggestion
The safety profile's namespace defines a set of concepts and data elements related to artifacts and their dependencies or links
```

bact commented on model/SystemsEngineering/Safety.md:
```diff
@@ -0,0 +1,21 @@
+to each other that are part of a safety related system's safety conformance documentation.
+These artifacts include the outputs of the safety engineering phases regarding planning, requirements analysis, system/software architecture, impelemenation, safety analysis and the verification tasks. The artifacts can be used to compile a standardized safety case document.
+
+
```
```suggestion

```

bact commented on model/SystemsEngineering/Classes/EvidenceRelationship.md:
```diff
@@ -0,0 +1,27 @@
+## Properties
+
+- evidenceUUID
+  - type: Core/ExternalIdentifier
```
```suggestion
  - type: /Core/ExternalIdentifier
```

bact commented on model/SystemsEngineering/Classes/RequirementVerification.md:
```diff
@@ -0,0 +1,39 @@
+## Properties
+
+- verificationUUID
+  - type: Core/ExternalIdentifier
```
```suggestion
  - type: /Core/ExternalIdentifier
```

nicpappler committed:
Update model/SystemsEngineering/Classes/RequirementVerification.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Delete model/SystemsEngineering/SystemsEngineering.md

Shelving the SystemsEngineering.md for v3.2

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Properties/verificationUUID.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Safety.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Safety.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Safety.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/Classes/EvidenceRelationship.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
adding the Functional Safety class diagram

nicpappler committed:
Merge branch 'add-functional-safety' of github.com:nicpappler/spdx-3-model-nicole into add-functional-safety

kestewart commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,19 @@
+
+## Summary
+
+The Safety Profile is designed to provide a standardized way of documenting and
```
```suggestion
The Functional Safety Profile is designed to provide a standardized way of documenting and
```

nicpappler committed:
Rename Safety.md to SystemsEngineering.md

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/SystemsEngineering/SystemsEngineering.md

Co-authored-by: Kate Stewart <13152682+kestewart@users.noreply.github.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

bact commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,19 @@
+
+## Description
+
+The safety profile's namespace defines a set of concepts and data elements related to artifacts and their dependencies or links
```
```suggestion
The Functional Safety profile is designed to provide a standardized way of documenting and
sharing information about artifacts created, verified and maintained during the safety lifecycle of a system.

## Description

The Functional Safety profile's namespace defines a set of concepts and data elements related to artifacts and their dependencies or links
```

bact commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,19 @@
+The safety profile's namespace defines a set of concepts and data elements related to artifacts and their dependencies or links
+to each other that are part of a safety related system's safety conformance documentation.
+These artifacts include the outputs of the safety engineering phases regarding planning, requirements analysis, system/software architecture, impelemenation, safety analysis and the verification tasks. The artifacts can be used to compile a standardized safety case document.
+## Metadata
```
```suggestion

## Metadata
```

bact commented on model/SystemsEngineering/SystemsEngineering.md:
```diff
@@ -0,0 +1,19 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# Safety
```
```suggestion
# Functional Safety
```

kestewart reviewed (approved):
We'll need to sort out the naming from SystemsEngineering --> FunctionalSafety;  but lets get this merged now, and sort it out when we can see the whole picture.

kestewart committed:
Update model/SystemsEngineering/SystemsEngineering.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Kate Stewart <13152682+kestewart@users.noreply.github.com>

bact committed:
Update model/SystemsEngineering/SystemsEngineering.md

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Update model/SystemsEngineering/SystemsEngineering.md

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact commented:
> We'll need to sort out the naming from SystemsEngineering --> FunctionalSafety; but lets get this merged now, and sort it out when we can see the whole picture.

OK. Let's merge for now and fix the failed CI later.
